### ISSUE TITLE
Share bot functionality?

### ISSUE THREAD

**Issue by nicoddemus** [rougeth/bottery#103]
I was thinking, some functionality like talking with Jenkins or GitHub could be shared with others so they can incorporate that functionality into their own bots, similar to how [Hubot](https://hubot.github.com/) works. Is this something we would like to have for `bottery`?

**Comment by rougeth** [comment]
I never used Hubot, didn't get the point. @nicoddemus could your provide an example?

**Comment by nicoddemus** [comment]
Sure.

Hubot is basically a hub, so you can install "scripts" into it that teach new commands. So for example, the bare `hubot` only understands `"hubot hello"` and replies with `"hey"`.

But then you install [hubot-jenkins](https://www.npmjs.com/package/hubot-jenkins) and now it understands:

* `hubot jenkins b ` - builds the job specified by jobNumber. List jobs to get number.
* `hubot jenkins build ` - builds the specified Jenkins job
* `hubot jenkins build , ` - builds the specified Jenkins job with parameters as key=value&key2=value2
* `hubot jenkins list ` - lists Jenkins jobs
* `hubot jenkins describe ` - Describes the specified Jenkins job
* `hubot jenkins last ` - Details about the last build for the specified Jenkins job

Or [hubot-business-cat](https://github.com/hubot-scripts/hubot-business-cat), which automatically listens for business jargons and posts memes.

In other words, the "scripts" are like plugins which teach your hubot new commands. 

In bottery terms, it would be as if I can install a `bottery-jenkins` plugin of sorts which would add a bunch of new `Patterns` to my bot automatically.

I think it would be nice to be able to program reusable "scripts" or "plugins" that we can share in the bottery community.

**Comment by rougeth** [comment]
That's nice! I think that doing something like Hubot should be easy on Bottery.
